File: battleship/cpp/Carrier.cpp
#include "Carrier.hpp"
#include <random>
#include <vector>
#include <string>


const int Carrier::CARRIER_SHOTS = 5;

const Carrier::Ship_sz Carrier::CARRIER_SIZE = 5;
  
const std::vector<std::string> Carrier::CARRIER_NAMES({"USS Wasp", "USS Hornet", "USS Lexington", "USS Saratoga", "USS Yorktown"});

void Carrier::init(){
  alive = true;
  shots = CARRIER_SHOTS;
  size = CARRIER_SIZE;
  name = CARRIER_NAMES[rand() % CARRIER_NAMES.size()];
}

File: battleship/java/Carrier.java
class Carrier extends Ship{
    
    String[] carrierNames = {"USS Wasp",
                             "USS Hornet",
                             "USS Essex",
                             "USS Lexington",
                             "USS Saratoga",
                             "USS Oriskany"};
    int carrierSize = 5;
    int carrierShots = 5;
    
    public Carrier(){
	init(carrierNames[(int) (Math.random() * carrierNames.length)], carrierSize, carrierShots);
    }
}
